src/cfg/File.h

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (C) 1996-2021 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#ifndef _SQUID__SRC_CFG_FILE_H
+#define _SQUID__SRC_CFG_FILE_H
+
+#include "base/Here.h"
+#include "cfg/forward.h"
+#include "sbuf/SBuf.h"
+
+#include <cstdio>
+#include <string>
+
+namespace Cfg
+{
+
+/**
+ * Class used to store required information for the current
+ * configuration file.
+ */
+class File
+{
+public:
+    File(const char *path) : filePath(path) {}
+    ~File();
+
+    /// \return true if the configuration file is open
+    bool isOpen() const { return bool(fd); }
+
+    /// \return the next line to be parsed from this file
+    SBuf nextLine();
+
+    /// \return the configuration file name and line number being processed
+    SourceLocation lineInfo() const { return SourceLocation("parsing", filePath.c_str(), lineNo); }
+
+    /// open and load contents from the file
+    void tryLoadFile();
+
+private:
+    std::string filePath; ///< The file path
+    FILE *fd = nullptr; ///< Pointer to the file.
+    SBufList fileData; ///< bytes read from this file in advance of parsing
+    int lineNo = 0; ///< Current line number being parsed
+
+    /// Whether this is a FIFO pipe instead of a file.
+    /// Enables '#line' notation support for this source.
```

Comment:
That is a feature addition which I am trying to avoid in this cleanup PR.

Reply:
Also, I did some code archaeology during the initial implementation and identified that this #line support was added by Henrik for admin using pipes. It has never been documented as supported in any way for actual files.

Reply:
AFAICT:

* parseOneConfigFile() supports `#line` for non-pipe files.
* parseOneConfigFile() parsers squid.conf and the files included via the `include` directive.
* 

Thus, whether properly documented or not, `#line` is currently honored for (important) non-pipe files. Folks are using this mechanism in their generated configuration files. If honoring `#line` creates (currently undisclosed) problems for your PR that you cannot solve, please discuss (hopefully, we can find a solution together). Otherwise, please continue to honor `#line` (at least where it was honored).

Reply:
Done in 673a789.